servers/rendering/renderer_rd/storage_rd/mesh_storage.cpp

```diff
@@ -1543,6 +1562,60 @@ void MeshStorage::_multimesh_set_mesh(RID p_multimesh, RID p_mesh) {
 	}
 	multimesh->mesh = p_mesh;
 
+	if (multimesh->indirect) { //Necessitates altering the command buffer to account for the new vertex count, going to assume at this time one surface.
+
+		Mesh *mesh = mesh_owner.get_or_null(p_mesh);
+		ERR_FAIL_NULL(mesh);
+		if (mesh->surface_count > 0) {
+			uint32_t index_count = 0;
+			uint32_t vertex_count = 0;
+			for (uint32_t i = 0; i < mesh->surface_count; i++) {
+				index_count += mesh->surfaces[i]->index_count;
```

Comment:
You shouldn't be adding all surfaces together. Each surface results in one draw call, so each surface needs its own entry in the command buffer. Then, when drawing, you use the `p_offset` argument to offset into the command buffer based on what surface you are drawing.

Reply:
I am unsure what you mean, would that mean that I technically need multiple command buffers for each surface? Each command buffer has to contain the vertex/index count of the mesh it's rendering, and if it is the command buffer for one surface, and it is supposed to be used to render another, it cannot be the same buffer unless I'm updating the vertex count on the command buffer with each draw call for each surface.

I am quite sure I am incorrect in what I am thinking here.

Reply:
It should be one command buffer, but that one command buffer will hold the data for multiple surfaces. Instead of holding one instance of the command struct, it will hold multiple. 

Then, when you draw, you just pass an offset into the command to tell the GPU which instance to use.

Reply:
For clarity, the offset I am talking about is the one in [`draw_list_draw_indirect`](https://docs.godotengine.org/en/latest/classes/class_renderingdevice.html#class-renderingdevice-method-draw-list-draw-indirect)

Instead of a command buffer that holds 4 (or 5) floats, you will create one that holds 5xnum_surfaces floats, the layout is basically:
```
    uint32_t    vertexCount0;
    uint32_t    instanceCount0;
    uint32_t    firstVertex0;
    uint32_t    firstInstance0;
    uint32_t    vertexCount1;
    uint32_t    instanceCount1;
    uint32_t    firstVertex1;
    uint32_t    firstInstance1;
    uint32_t    vertexCount2;
    uint32_t    instanceCount2;
    uint32_t    firstVertex2;
    uint32_t    firstInstance2;
```

Then when drawing:
```
//surface 0
draw_list_draw_indirect(draw_list, false, command_buffer, 0, 1, 16);
//surface 1
draw_list_draw_indirect(draw_list, false, command_buffer, 16, 1, 16);
//surface 2
draw_list_draw_indirect(draw_list, false, command_buffer, 32, 1, 16);
```

Reply:
Affirm I'll look into this after work today.

Thank you for taking the time to explain it, this may require the documentation to be a bit more verbose, and possibly just a straight up example project, as compute shaders editing the command buffer of a multi surface mesh will have to account for updating the instance count in multiple places, and it has to account for if the mesh is indexed or not.

Reply:
Got it sorted, a little late to the party but got it sorted.